zulip/zulip
# mention: Replace 'wildcards' with 'stream_wildcards'.

**Summary**

[#25828 (comment)](https://github.com/zulip/zulip/pull/25828#issuecomment-1581675528)
> The 4th commit might be worth extracting to a separate PR to make it easier to verify we're handling https://github.com/zulip/zulip/pull/25828#discussion_r1222293742 correctly; I think the rename is a good change but we want to be really careful, because bugs of that form can require manual intervention to handle in production.

---
This PR replaces the `wildcard` keyword in the codebase with `stream_wildcard` at some places for better readability, as we plan to introduce `topic_wildcards` as a part of the '@topic mention' project.

Currently, **wildcards = ["all", "everyone", "stream"]** which is an alias to mention everyone in the stream, hence better renamed as **stream_wildcards**.

Eventually, we will have:
* `stream_wildcard` as an alias to mention everyone in the stream.
* `topic_wildcard` as an alias to mention everyone in the topic.
* `wildcard` refers to 'stream_wildcard' and 'topic_wildcard' as a whole.
---
**Questions:**

* [#25828 (comment)](https://github.com/zulip/zulip/pull/25828#discussion_r1222291166) -- Doubt related to django migrations.
---
**Manual Testing**

After addressing this important [review comment](https://github.com/zulip/zulip/pull/25828#discussion_r1222293742).

I did manual testing by adding the translation code (`TODO/compatibility`) in `event_queue.py` and keeping the field in the event sent by `message_send` and `message_edit` codepath as `wildcard_mention_user_ids` (instead of `stream_wildcard_mention_user_ids`).

Used wildcard mentions in messages with no error shown on the console.

With the earlier code (no translation code added), it was indeed giving unexpected results.



---
<details>
<summary>Self-review checklist</summary>





      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.


Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).


Completed manual review and testing of the following:

</details>
